### PR TITLE
Made it work for actual profiles

### PR DESCRIPTION
**Pull request by roxyboy**
The code works now for actual profiles with rigid lid and flat ocean approximation.

### CODE REVIEW

**Comment by roxyboy** [comment]
Profile (black line: actual profile, red line: exponential fit)
screen shot 2016-05-29 at 12 04 48

Result (exponential fit w/ Ah=10):
screen shot 2016-05-29 at 15 39 34

Result (actual profile w/ Ah=0):
screen shot 2016-05-29 at 15 39 46

Result (actual profile w/ Ah=10):
screen shot 2016-05-29 at 15 39 59

**Comment by rabernat** [comment]
@roxyboy this is fantastic! Great progress!

Could you please add some additional tests corresponding to the examples you provided above?

Also, please remove the file `test_oceanmodes_on_OCCA.html`. If you want to add an example ipython notebook, then add the notebook directly (.ipynb file), rather than an html file. Put it in a folder called `examples`.

**Comment by rabernat** [comment]
Why is it necessary to store all of the OCCA binary files in the repo? This really increases the size of the repo, and they are available elsewhere. As far as I can tell, they aren't used in any of the tests.

Also, do you know why the travis build didn't work?

File: oceanmodes/test/test_modes.py [hunk 1]
```diff
@@ -264,3 +266,114 @@ def test_Eady(self, atol=5e-2, nz=20, Ah=0.):
...
+        
+        nz = len(z)
+
+        ###################
+        # make sure we got the right number of modes
+        # NOPE! don't want all the modes and scipy.sparse.linalg.eigs won't
+        # compute them
+        # self.assertEqual(nz, len(def_radius))
+        # make sure the modes themselves have the right structure
+        ###################
```
Comment (rabernat) [inline oceanmodes/test/test_modes.py]: Can we remove these redundant comments?

File: oceanmodes/test/test_modes.py [hunk 2]
```diff
@@ -264,3 +266,114 @@ def test_Eady(self, atol=5e-2, nz=20, Ah=0.):
...
+        self.assertTrue(np.all( np.diff(
+                    growth_rate.reshape((growth_rate.shape[0], len(k)*len(l))).imag.max(axis=1) ) <= 0.),
+        #self.assertTrue(np.all( max_growth_rate == 0.),
+            msg='imaginary part of modes should be descending')
+
+        #nmodes = len(def_radius)
+        #zero_crossings = np.abs(np.diff(np.sign(bc_modes), axis=0)/2).sum(axis=0)
+        #self.assertListEqual(list(range(nmodes)), list(zero_crossings),
+        #    msg='modes should have the correct number of zero crossings')
+
```
Comment (rabernat) [inline oceanmodes/test/test_modes.py]: And these?

File: oceanmodes/test/test_modes.py [hunk 3]
```diff
@@ -264,3 +266,114 @@ def test_Eady(self, atol=5e-2, nz=20, Ah=0.):
...
+        mode_amplitude1 = (np.absolute(vertical_modes[:, 0])**2).sum(axis=0)
+        self.assertTrue(np.allclose(1., mode_amplitude1),
+            msg='mode1 should be normalized to amplitude of 1 at all horizontal wavenumber points')
+
+        mode_amplitude2 = (np.absolute(vertical_modes[:, 1])**2).sum(axis=0)
+        self.assertTrue(np.allclose(1., mode_amplitude2),
+            msg='mode2 should be normalized to amplitude of 1 at all horizontal wavenumber points')
```
Comment (rabernat) [inline oceanmodes/test/test_modes.py:371]: What about testing for the actual values of the growth rates?

Comment (roxyboy) [inline oceanmodes/test/test_modes.py]: The OCCA data files were also added by mistake. They have already been deleting from git by the command 'git rm --cached' so I currently don't know how to delete the binary files. Maybe we could delete them from the Github web-based edit after we merge.

I do not know why travis build did not work.

As for the actual values of the growth rates, the profiles used do not have an analytical solution so I don't have anything to compare them with.

**Comment by roxyboy** [comment]
The OCCA data files were also added by mistake. They have already been deleted from git by the command 'git rm --cached' so I currently don't know how to delete the binary files. Maybe we could delete them from the Github web-based edit after we merge.

I do not know why travis build did not work.

As for the actual values of the growth rates, the profiles used do not have an analytical solution so I don't have anything to compare them with.

File: oceanmodes/test/test_modes.py (hunk 3 above)
Comment (rabernat) [inline oceanmodes/test/test_modes.py:371]: > As for the actual values of the growth rates, the profiles used do not have an analytical solution so I don't have anything to compare them with.

You have the output of your current code to compare with. That way, if future code changes alter the output of the routine, or if the results are different on different platforms, we will know.

**Comment by rabernat** [comment]
Just a reminder of what we discussed yesterday:
- [x] move the consistency checks of the input profiles into `_maybe_truncate_above_topography`
- [ ] add tests for different input profiles, including raising appropriate errors
- [x] add test for actual growth rates

File: oceanmodes/baroclinic.py [hunk 4]
```diff
@@ -8,26 +8,37 @@
...
+        # the above check was redundant with this one
+        if fm.mask[-1] and np.diff(fm.mask).sum() != 1:
+            raise ValueError('topographic mask should be monotonic')
+        
+        if ndata > 1:    
+            if n == 0:
+                fout = np.zeros((ndata, len(fm.compressed())))
+            fout[n] = fm.compressed()
+        else:
+            fout = fm.compressed()
```
Comment (rabernat) [inline oceanmodes/baroclinic.py:45]: Should we check to be sure that all the input profiles have the same mask?

**Comment by rabernat** [comment]
Great! Let's do it!

**Pull request merged**